### ISSUE TITLE
Strange "Cannot use namespace 'Foo' as a type" error with dummy `declare module "foo";` declaration

### ISSUE THREAD

**Issue by mattmccutchen** [Microsoft/TypeScript#27311]
Maybe this is a question, but it has come up multiple times on Stack Overflow without a good answer, so I'm asking it here.

**TypeScript Version:**  master (03af107)

**Search Terms:** "cannot use namespace" "as a type" TS2709 "declare module"

**Code**
```ts
declare module "foo";

declare module "bar" {
    import { Foo } from "foo";
    let x: Foo;
}
```

**Expected behavior:** An error that makes sense, or no error?

**Actual behavior:** `error TS2709: Cannot use namespace 'Foo' as a type.`

**Playground Link:** [link](https://www.typescriptlang.org/play/#src=declare%20module%20%22foo%22%3B%0D%0A%0D%0Adeclare%20module%20%22bar%22%20%7B%0D%0A%20%20%20%20import%20%7B%20Foo%20%7D%20from%20%22foo%22%3B%0D%0A%20%20%20%20let%20x%3A%20Foo%3B%0D%0A%7D)

**Related Issues:** None found

**Comment by RyanCavanaugh** [comment]
It is technically correct from the perspective of how the compiler is implemented 🙃 

In the specific case where we imported an `any`-like value in an ambient context and then try to use it as a type, we could say something like:

 > The name 'Foo' does not refer to a known type

Open to bikeshedding

**Comment by tjcarter1982** [comment]
Im having the same issue with trying to use create-react-app with typescript ver 3.2.1 and loona

**Comment by cmdcolin** [comment]
This error was happening to me when I accidentally had `declare module "mymodule"` but then actually was using `import {MyInterface} from 'mymodule'` and this caused any usage of MyInterface after that resulted in the "Cannot use namespace 'MyInterface' as a type" and "Property 'myprop' of exported interface has or is using private name 'MyInterface'." on the same line

I had to delete the `declare module "mymodule"` line and then it worked. This was very confusing to me so I'd add a  +1 for a better error message if possible

**Comment by FireStormHR** [comment]
> This error was happening to me when I accidentally had `declare module "mymodule"` but then actually was using `import {MyInterface} from 'mymodule'` and this caused any usage of MyInterface after that to result in the errors "Cannot use namespace 'MyInterface' as a type" and "Property 'myprop' of exported interface has or is using private name 'MyInterface'." on the same line
> 
> I had to delete the `declare module "mymodule"` line and then it worked. This was very confusing to me so I'd add a +1 for a better error message if possible

@cmdcolin having your module declared is necessary when using typescript and having `noImplicitAny `turned on in the `ts.config`. To solve this **namespace** error while keeping your declaration you can put `typeof` in front of the place where you are using the namespace.

**Comment by cmdcolin** [comment]
@FireStormHR the main thing for me is that when a module has no typings, I just say "define namespace 'mymodule'" to just sort of add "any" typings to all of mymodule. After awhile I created actual typescript typings for 'mymodule' though, and then after doing that, trying to import those types caused the error this github issue talks about. Deleting the "define module 'mymodule'" from my declare.d.ts fixed it. It's kind of tricky since declare.d.ts isn't a file I look at too often.  I am not familiar with using namespaces much beyond the above usage so I am not actually familiar with the situation you refer to

**Comment by qwabra** [comment]
welcome `TypeScript v3.8` with `import type {} from ''`

**Comment by andreyluiz** [comment]
In my case, the errors were off when I renamed a React component file from `.ts` to `.tsx` extension. TS seems to really care about the extension when it is a React component.

**Comment by Christopher-Hayes** [comment]
In my case just removing `.ts` entirely fixed the error strangely enough. I thought omitting the extension didn't matter, but I guess sometimes it does.
For example:
`import { ObjectName } from '@/path/to/file.ts'` <-- before
`import { ObjectName } from '@/path/to/file'` <-- after

**Comment by kulkarnipradnyas** [comment]
create a tsconfig.json and  "compilerOptions": {
    "outDir": "./dist/",
    "noImplicitAny": true,
    "module": "es6",
    "target": "es5",
    "jsx": "react",
    "allowJs": true,
    "sourceMap": true,
    "moduleResolution": "node",
    "allowSyntheticDefaultImports":true
  },
allowSyntheticDefaultImports particular thing will solve your problem

**Comment by bombillazo** [comment]
> 
> 
> > This error was happening to me when I accidentally had `declare module "mymodule"` but then actually was using `import {MyInterface} from 'mymodule'` and this caused any usage of MyInterface after that to result in the errors "Cannot use namespace 'MyInterface' as a type" and "Property 'myprop' of exported interface has or is using private name 'MyInterface'." on the same line
> > I had to delete the `declare module "mymodule"` line and then it worked. This was very confusing to me so I'd add a +1 for a better error message if possible
> 
> @cmdcolin having your module declared is necessary when using typescript and having `noImplicitAny `turned on in the `ts.config`. To solve this **namespace** error while keeping your declaration you can put `typeof` in front of the place where you are using the namespace.

This did it for me!

**Comment by denkhis** [comment]
> In my case just removing `.ts` entirely fixed the error strangely enough. I thought omitting the extension didn't matter, but I guess sometimes it does. For example: `import { YourType } from '@/path/to/file.ts'` <-- before `import { YourType } from '@/path/to/file'` <-- after

thanks!
it worked for me
